src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -74,14 +95,14 @@ impl QemuInner {
         todo!()
     }
 
-    /// TODO: using a single hardcoded CID is clearly not adequate in the long
-    /// run. Use the recently added VsockConfig infrastructure to fix this.
     pub(crate) async fn get_agent_socket(&self) -> Result<String> {
         info!(sl!(), "QemuInner::get_agent_socket()");
-        Ok(format!(
-            "{}://{}:{}",
-            VSOCK_SCHEME, VSOCK_AGENT_CID, VSOCK_AGENT_PORT
-        ))
+        let guest_cid = match &self.get_agent_vsock_dev() {
+            Some(device) => device.config.guest_cid,
+            None => return Err(anyhow!("uninitialized agent vsock".to_owned())),
+        };
```

Comment:
Frankly I'd like to keep it this way, at least for now, because
- arguably `VsockConfig::guest_cid` _is_ the canonical place for CID to live
- as you say yourself, it's not quite clear where else to store it
- I'd really like to put effort into preventing `QemuInner` from becoming a grab bag of anything that didn't fit anywhere else, mixing levels of abstraction wildly (a bit like the `Config` structure in govmm which is a collection of high-level concepts like `Machine`, `Kernel` or `Memory` along with loose strings, ints and arrays)
- I believe that runtime complexity is not a concern here as I think that `get_agent_socket()` is called exactly once (this would be different if CID was being retrieved over and over again).  Also, there won't be hundreds of devices to make the lookup time a problem.
- if I remember correctly the idea of looking up a device to retrieve a piece of information from it wasn't actually mine in the first place, I think it comes from another hypervisor driver (probably dragonball or clh?) so it doesn't seem a bad idea to do stuff in a similar manner across hv drivers, where applicable

Ultimately I'm far from suggesting that this is the best solution forever and it's quite likely a different way might present itself during the course of finishing the qemu driver but until then I'd frankly prefer to keep it this way.  Would that be acceptable?

Reply:
Yeah, we can discuss it later in development and get a better one.